hack/docker-push.sh

```diff
@@ -56,26 +56,25 @@ elif [[ "$triggeredBy" == "tags" ]]; then
     TAG=$(echo $GITHUB_REF | cut -d / -f 3)
 fi
 
-if [[ "$BRANCH" == "main" ]]; then
-    VERSION="$BRANCH"
-elif [[ ! -z "$TAG" ]]; then
+if [[ ! -z "$TAG" ]]; then
+    echo "We're building tag $TAG"
     # Explicitly checkout tags when building from a git tag.
     # This is not needed when building from main
     git fetch --tags
     # Calculate the latest release if there's a tag.
     highest_release
     VERSION="$TAG"
 else
-    echo "We're not on main and we're not building a tag, exit early."
-    exit 0
+    echo "We're on branch $BRANCH"
+    VERSION="$BRANCH"
```

Comment:
IMO we should only push image for main and release-*

Reply:
The branches are specified in `push.yaml` which only cover branch `main` and `release-**`

Reply:
right